### PR TITLE
[thread-cert] add case 5.8.4 SecurityPolicyTLV using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**
add security policy set support in dataset

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+#
+# Notes: Due to the packet parsing compatiable issue for supporting Thread 1.2
+#        and 1.1, the security policy values can be fetched only in the unknown
+#        field.
+#
+# Test Topology:
+# -------------
+#  Leader
+#     |
+# Commissioner
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py:59]: add commissioner2 and ThreadNode to the topology?

File: tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+#
+# DUT Types:
+# ----------
+#  Leader
+
+
+class Cert_5_8_04_SecurityPolicyTLV(thread_cert.TestCase):
+    SUPPORT_NCP = False
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py:67]:
```suggestion
    SUPPORT_NCP = False
    USE_MESSAGE_FACTORY = False
```
nit

File: tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+            filter(lambda p: p.thread_meshcop.tlv.state == 1).\
+            must_next()
+
+        # Step 16: The DUT MUST send beacon response frames.The beacon payload MUST
+        #          either be empty OR the payload format MUST be different from the
+        #          Thread Beacon payload The Protocol ID and Version field values
+        #          MUST be different from the values specified for the Thread beacon
+        #          (Protocol ID= 3, Version = 2)
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_beacon().\
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py:345]: can we check the beacon payload?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

File: tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py (hunk 3 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py:349]: add thread beacon fields and checks

File: tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_5_8_04_SecurityPolicyTLV.py]: added

**Review by jwhui — APPROVED** [review]

**Pull request merged**